src/bytes.rs

```diff
@@ -804,10 +805,9 @@ impl From<Vec<u8>> for Bytes {
             return Bytes::new();
         }
 
-        let slice = vec.into_boxed_slice();
+        let slice = ManuallyDrop::new(vec.into_boxed_slice());
         let len = slice.len();
         let ptr = slice.as_ptr();
-        drop(Box::into_raw(slice));
```

Comment:
How does this compare to just doing this?
```Rust
let ptr = Box::into_raw(slice) as *const _;
```

Reply:
Where the `as` here is doing the wide-to-thin ptr conversion?

Yeah something like that should also work. The important part is not using `slice` any more after `ptr` got created.

Reply:
I'm mainly worried about the `as_ptr` creating the pointer through a `&self` instead of `&mut self`, but that may not be a problem since this is for `Bytes` where we can't modify it anyway.

Reply:
That's fair. I updated my patch.